src/Stryker.Core/Stryker.Core/Helpers/RoslynHelper.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Stryker.Core.Mutants;
+
+namespace Stryker.Core.Helpers
+{
+    internal static class RoslynHelper
+    {
+        // gets the return type of a method (incl. constructor, destructor...)
+        public static TypeSyntax ReturnType(this BaseMethodDeclarationSyntax baseMethod)
+        {
+            return baseMethod switch
+            {
+                ConstructorDeclarationSyntax _ => null,
+                ConversionOperatorDeclarationSyntax _ => null,
+                DestructorDeclarationSyntax _ => null,
+                MethodDeclarationSyntax method => method.ReturnType,
+                OperatorDeclarationSyntax oper => oper.ReturnType,
+                _ => null
+            };
+        }
+
+        public static bool NeedsReturn(this BaseMethodDeclarationSyntax baseMethod)
+        {
+            return baseMethod switch
+            {
+                MethodDeclarationSyntax method => !(method.ReturnType is PredefinedTypeSyntax predefinedType &&
+                                                    predefinedType.Keyword.IsKind(SyntaxKind.VoidKeyword)),
+                OperatorDeclarationSyntax _ => true,
+                ConversionOperatorDeclarationSyntax _=> true,
+                _ => false
+            };
+        }
+
+        public static T InjectMutation<T>(this T original, Mutation mutation) where T:SyntaxNode
+        {
+            if (!original.Contains(mutation.OriginalNode))
+            {
+                throw new InvalidOperationException($"Cannot inject mutation '{mutation.ReplacementNode}' in {original}.");
```

Comment:
```suggestion
                throw new InvalidOperationException($"Cannot inject mutation '{mutation.ReplacementNode}' in {original} because this node and mutation original node are not the same.");
```